packages/cursorless-neovim-e2e/src/suite/recorded.neovim.test.ts

```diff
@@ -0,0 +1,392 @@
+  const marks = undefined;
+
+  if (fixture.finalState?.clipboard == null) {
+    excludeFields.push("clipboard");
+  }
+
+  for (const storedTargetKey of storedTargetKeys) {
+    const key = `${storedTargetKey}Mark` as const;
+    if (fixture.finalState?.[key] == null) {
+      excludeFields.push(key);
+    }
+  }
+
+  // FIXME Visible ranges are not asserted, see:
+  // https://github.com/cursorless-dev/cursorless/issues/160
+  const { visibleRanges, ...resultState } = await takeSnapshot(
+    excludeFields,
+    [],
+    spyIde.activeTextEditor!,
+    spyIde,
+    marks,
+    false,
+  );
+
+  const rawSpyIdeValues = spyIde.getSpyValues(fixture.ide?.flashes != null);
+  const actualSpyIdeValues =
+    rawSpyIdeValues == null
+      ? undefined
+      : spyIDERecordedValuesToPlainObject(rawSpyIdeValues);
+
+  if (shouldUpdateFixtures()) {
+    const outputFixture: TestCaseFixtureLegacy = {
+      ...fixture,
+      finalState: resultState,
+      returnValue,
+      fallback,
+      ide: actualSpyIdeValues,
+      thrownError: undefined,
+    };
+
+    await fsp.writeFile(file, serializeTestFixture(outputFixture));
+  } else {
+    if (fixture.thrownError != null) {
+      throw Error(
+        `Expected error ${fixture.thrownError.name} but none was thrown`,
+      );
+    }
+
+    assert.deepStrictEqual(
+      resultState,
+      fixture.finalState,
+      "Unexpected final state",
+    );
+
+    assert.deepStrictEqual(
+      returnValue,
+      fixture.returnValue,
+      "Unexpected return value",
+    );
+
+    assert.deepStrictEqual(
+      fallback,
+      fixture.fallback,
+      "Unexpected fallback value",
+    );
+
+    assert.deepStrictEqual(
+      omitByDeep(actualSpyIdeValues, isUndefined),
+      fixture.ide,
+      "Unexpected ide captured values",
+    );
+  }
+}
+
+const failingFixtures = [
+  // actual finalState.selections.anchor is -1 compared to expected (other fixture.command.action.name == "insertCopyBefore" tests pass fine)
+  "recorded/actions/cloneToken4",
+  "recorded/actions/cloneUpToken4",
+  // -> Error: nvim_execute_lua: Cursor position outside buffer
+  "recorded/compoundTargets/chuckStartOfBlockPastStartOfFile",
+  // actual finalState.selections.anchor is -1 compared to expected
+  "recorded/implicitExpansion/chuckCoreThat",
+  "recorded/implicitExpansion/chuckLeadingThat",
+  "recorded/marks/chuckNothing",
+  // -> wrong fixture.finalState.selections
+  "recorded/implicitExpansion/cloneThat2",
+  "recorded/implicitExpansion/cloneThis",
+  "recorded/implicitExpansion/cloneThis2",
+];
+
+function isFailingFixture(name: string, fixture: TestCaseFixtureLegacy) {
+  const action =
+    typeof fixture.command.action === "object"
+      ? fixture.command.action.name
+      : fixture.command.action;
+
+  // "recorded/actions/insertEmptyLines/puffThis*" -> wrong fixture.finalState.selections and fixture.thatMark.contentRange
+  if (action === "insertEmptyLinesAround") {
```

Comment:
this series of `if` statements would be much more compact as a `switch` statement